op-batcher/batcher/espresso.go

```diff
@@ -321,3 +327,96 @@ func (l *BatchSubmitter) fetchBlock(ctx context.Context, blockNumber uint64) (*t
 
 	return block, nil
 }
+
+func (l *BatchSubmitter) waitForFinality(submittedTxInfo espressoCommon.TransactionQueryData) error {
+	rawHeader, err := l.Espresso.FetchRawHeaderByHeight(l.shutdownCtx, submittedTxInfo.BlockHeight)
+	if err != nil {
+		return err
+	}
+
+	var header espressoCommon.HeaderImpl
+	err = json.Unmarshal(rawHeader, &header)
+	if err != nil {
+		return fmt.Errorf("could not unmarshal header from bytes")
+	}
+
+	height := header.Header.GetBlockHeight()
+
+	timer := time.NewTimer(finalityTimeout)
+	defer timer.Stop()
+
+	ticker := time.NewTicker(finalityCheckInterval)
+	defer ticker.Stop()
+
+	var snapshot espressoCommon.BlockMerkleSnapshot
+
+Loop:
+	for {
+		select {
+		case <-ticker.C:
+			res, err := l.EspressoLightClient.FetchMerkleRoot(height, nil)
```

Comment:
Since we are going to remove the use of the light client (task: https://app.asana.com/1/1208976916964769/project/1209392461754458/task/1209928931628267?focus=true), should we replace/remove this line?